lib/xform.js

```diff
@@ -26,6 +29,38 @@ var XFormState = State.extend(function (self, name, opts) {
         user or ``null`` if no address was chosen. If ``next`` is ``null`` or
         not defined, the state machine will be left in the current state.
         See :meth:`State.set_next_state`. Defaults to ``null``
+    :param string opts.xforms_service_url:
+        The URL where the XForms service is located. Defaults to
+        ``prd-vumi-xforms.aws.prk-host.net``.
+    :param string opts.xforms_service_error_message:
+        The error message sent back to the user if the XForms service is
+        inaccessable. Defaults to
+        ``Error contacting the xforms service``.
+    :param string opts.xform:
```

Comment:
If we allowed `xform` to be either a constant string, a function that returned a string, or a function that returned a promise (and removed `xform_url`, `xform_url_username`, `xform_url_password`), we could increase the flexibility of how the xform is loaded and at the same time make this state simpler. Thoughts?